pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -645,7 +646,21 @@ func (c *Controller) GetIstioServiceAccounts(hostname model.Hostname, ports []in
 		saArray = append(saArray, sa)
 	}
 
-	return saArray
+	var accounts []string
+	accounts = append(accounts, saArray...)
```

Comment:
That means both cluster.local and trustDomain will be added - seems like a security issue.